jhove-ext-modules/src/main/java/org/ithaka/portico/jhove/module/EpubModule.java

```diff
@@ -0,0 +1,567 @@
+        infoProperties.add(generateProperty(PROPNAME_CONTRIBUTOR, report.getContributors()));
+        infoProperties.add(generateProperty(PROPNAME_DATE, report.getDate()));
+        infoProperties.add(generateProperty(PROPNAME_PUBLISHER, report.getPublisher()));
+        infoProperties.add(generateProperty(PROPNAME_SUBJECTS, report.getSubjects()));
+        infoProperties.add(generateProperty(PROPNAME_RIGHTS, report.getRights()));
+        infoProperties.remove(null);
+
+        properties.add(generateProperty(PROPNAME_INFO, infoProperties));
+
+        Set<Property> fontList = generateFontProps(report.getEmbeddedFonts(), true);
+        fontList.addAll(generateFontProps(report.getRefFonts(), false));
+        properties.add(generateProperty(PROPNAME_FONTS, fontList));
+
+        properties.add(generateProperty(PROPNAME_REFERENCES, report.getReferences()));
+        properties.add(generateProperty(PROPNAME_LOCALRESOURCES, report.getLocalResources()));
+        properties.add(generateProperty(PROPNAME_REMOTERESOURCES, report.getRemoteResources()));
+        properties.add(generateProperty(PROPNAME_MEDIATYPES, report.getMediaTypes()));
+
+        for (String feature : report.getFeatures()) {
+            properties.add(generateProperty(feature, true));
+        }
+
+        properties.remove(null);
+
+        return properties;
+    }
+
+    /**
+     * Returns true if a message is either (a) a fatal error or (b) non-fatal error
+     * whose error id is listed in NOTWELLFORMED_ERRCODES
+     *
+     * @param id
+     * @return
+     */
+    private boolean triggerNotWellFormed(CheckMessage msg) {
+        if (msg.getSeverity().equals(Severity.FATAL)) {
+            return true;
+        }
+        if (msg.getSeverity().equals(Severity.ERROR)) {
+            for (String s : NOTWELLFORMED_ERRCODES) {
+                if (msg.getID().startsWith(s)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Generates the a set of font properties to be included in the JHOVE report
+     *
+     * @param fonts    List of fonts found in the EPUB
+     * @param fontFile true if the font file is embedded in the EPUB
+     * @return
+     */
+    private Set<Property> generateFontProps(Set<String> fonts, boolean fontFile) {
+
+        Set<Property> fontPropertiesList = new HashSet<Property>();
+
+        if (fonts != null && fonts.size() > 0) {
+            for (String font : fonts) {
+                Set<Property> fontProps = new HashSet<Property>();
+                fontProps.add(generateProperty(PROPNAME_FONTNAME, font));
+                fontProps.add(generateProperty(PROPNAME_FONTFILE, fontFile));
+
+                fontPropertiesList.add(generateProperty(PROPNAME_FONT, fontProps));
+            }
+        }
+        fontPropertiesList.remove(null);
+
+        return fontPropertiesList;
+
+    }
+
+    /**
```

Comment:
These generate property methods appear to be useful, but they're also quite generic and might be useful to other module devs. While I don't propose moving them yet it looks like a useful maintenance / refactoring task for a later date.